src/graph/executor/query/AppendVerticesExecutor.cpp

```diff
@@ -26,10 +26,12 @@ StatusOr<DataSet> AppendVerticesExecutor::buildRequestDataSet(const AppendVertic
 
 folly::Future<Status> AppendVerticesExecutor::appendVertices() {
   SCOPED_TIMER(&execTime_);
-
   auto *av = asNode<AppendVertices>(node());
-  StorageClient *storageClient = qctx()->getStorageClient();
+  if (FLAGS_optimize_appendvertices && av != nullptr && av->props() == nullptr) {
```

Comment:
This will lead error if dangle edges presents.

Reply:
will be turned on when testing

Reply:
If so, I think you could eliminate AppendVertices operator in optimizer.

Reply:
We don't have test cases for dangling edges now. If this operator returns directly, the behavior is different from before.

Reply:
I also agree with @Shylock-Hg  that if property Pruning is a deterministic optimization, isn't it better practice to handle it in the optimizer?

Reply:
there will be a problem of dangle edges. Currently it is manually controlled, and it is usually turned off. It is only turned on when the LDBC test is performed